### PR TITLE
Fixed issue 117: pure function calls on union receivers crash the verifier

### PR DESCRIPTION
**Pull request by fabiopakk**
This is the pull request regarding issue 117. Additionally to the test case added from the issue report, other 3 test cases were added to fully exercise all possible inputs the modified function is expected to handle.

---
This change is [Reviewable](https://reviewable.io/reviews/marcoeilers/nagini/122)

### CODE REVIEW

**Comment by marcoeilers** [comment]
It seems that the implementation always tries to assign the results of the individual calls to a variable. That's a problem because that would not be possible in a pure context (e.g., inside a pure function). If all options for a call are pure functions, it would be better if the entire resulting translation was an expression, not a statement. 

As an example, if you add a ``@Pure`` annotation to one of the functions in the new test, Nagini reports an invalid program:

```
@Pure
def test_union_pure_functions(u: Union[E, F]) -> int:
    Ensures(Result() == 5 or Result() == 7)
    return u.some_int()
```

**Comment by fabiopakk** [comment]
Good point, I'll handle this.

**Comment by fabiopakk** [comment]
Code changed to support `@Pure` as requested and also added the above example as test case. Please check if those modifications are compliant.

File: src/nagini_translation/translators/call.py [hunk 1]
```diff
@@ -857,34 +858,66 @@ def _inline_call(self, method: PythonMethod, node: ast.Call, is_super: bool,
     def translate_method_call_in_union(self, arg_stmts: List[Stmt], args: List[Expr],
                                        arg_types: List[PythonType], rectype: UnionType,
                                        node: ast.Call, ctx: Context,
-                                       impure) -> StmtsAndExpr:
+                                       impure: bool) -> StmtsAndExpr:
         """
         Translate a method call, when the receiver is of type union, into an if-then-else
```
[the remark this answers is not in the document]
Comment (fabiopakk) [reply src/nagini_translation/translators/call.py]: Fair enough. Renamed the function and updated the docstring. Indeed, semantics changed. I'll merge into master and close issue 117, ok?

**Comment by fabiopakk** [comment]
Branch renamed to fabiopakk_issue_117_pure_functions.

**Pull request merged**